datacube_alchemist/worker.py

```diff
@@ -228,3 +237,49 @@ def execute_pickled_task(pickled_task):
     _LOG.info("Found task to process: {}".format(task))
     execute_task(task)
     s3ul.upload_if_needed()
+
+
+def deterministic_uuid(task, algorithm_version=None, **other_tags):
+    if algorithm_version is None:
+        transform_info = get_transform_info(task.settings.specification.transform)
+        algorithm_version = transform_info['version_major_minor']
+    if 'dataset_version' not in other_tags:
+        try:
+            other_tags['dataset_version'] = task.settings.output.metadata['dataset_version']
+        except KeyError:
+            msg = 'dataset_version not set and '
+            msg += 'not used to generate deterministic uuid'
+            _LOG.info(msg)
```

Comment:
Why are we building a `msg` and then logging it instead of just logging the message?